network/libp2p_impl.go

```diff
@@ -186,7 +212,16 @@ func (gsnet *libp2pGraphSyncNetwork) ConnectTo(ctx context.Context, p peer.ID) e
 
 // handleNewStream receives a new stream from the network.
 func (gsnet *libp2pGraphSyncNetwork) handleNewStream(s network.Stream) {
+	var p peer.ID
+
 	defer s.Close()
+	defer func() {
+		if rerr := gsnet.panicHandler(recover()); rerr != nil {
+			log.Debugf("graphsync net handleNewStream recovered error from %s error: %s", s.Conn().RemotePeer(), rerr)
+			_ = s.Reset()
+			go gsnet.receiver.ReceiveError(p, rerr)
```

Comment:
@hannahhoward I wouldn't mind a sanity-check that this is appropriate behaviour on a panic, it's roughly the same as for a standard error from the message handler.

Reply:
.. there's potential for `p` to be unset at this call since it's only set after the `FromMsgReader` call below; alternative might be to move the `p =` up one line to have a better chance of it being set (it could also be unset because of a panic somewhere else! like `NewVarintReaderSize()`.